tests/validation/tests/v3_api/test_certificate.py

```diff
@@ -0,0 +1,571 @@
+                    certs=rancher_cert
+                )
+                assert e.value.error.status == 403
+                assert e.value.error.code == 'Forbidden'
+        else:
+            certificate_valid = p_client.create_namespaced_certificate(
+                name=cert_name_1, key=rancher_private_key, certs=rancher_cert,
+                namespaceId=ns['name']
+            )
+            assert certificate_valid.issuer == 'Let\'s Encrypt Authority X3'
+            certificate_allns_valid = p_client.create_certificate(
+                name=cert_name_2, key=rancher_private_key,
+                certs=rancher_cert
+            )
+            assert certificate_allns_valid.issuer == 'Let\'s Encrypt ' \
+                                                     'Authority X3'
+            # Delete the resources
+            p_client.delete(certificate_valid)
+            p_client.delete(certificate_allns_valid)
+
+    @if_test_rbac
+    @pytest.mark.parametrize("role", rbac_role_list)
+    def test_list_certificate(self, role):
+        """
+        Test steps:
+        1. Create certificate for single and all namespace for all role as
+        cluster owner
+        2. List the created certificate for all roles
+        2. Delete the certificates
+        """
+        c_owner_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        token = rbac_get_user_token_by_role(role)
+        project = rbac_get_project()
+        ns = rbac_get_namespace()
+        p_client = get_project_client_for_token(project, token)
+        p_client_owner = get_project_client_for_token(project,
+                                                            c_owner_token)
+        cert_name_1 = random_test_name("cert-rbac")
+        cert_name_2 = random_test_name("cert-rbac")
+        certificate_valid = p_client_owner.create_namespaced_certificate(
+            name=cert_name_1, key=rancher_private_key, certs=rancher_cert,
+            namespaceId=ns['name']
+        )
+        certificate_allns_valid = p_client_owner.create_certificate(
+            name=cert_name_2, key=rancher_private_key,
+            certs=rancher_cert
+        )
+        if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
+            cert_count = p_client.list_namespaced_certificate(name=cert_name_1)
+            cert_count_1 = p_client.list_certificate(name=cert_name_2)
+            assert len(cert_count) == 0
+            assert len(cert_count_1) == 0
+        else:
+            cert_count = p_client.list_namespaced_certificate(name=cert_name_1)
+            cert_count_1 = p_client.list_certificate(name=cert_name_2)
+            assert len(cert_count) > 0
+            assert len(cert_count_1) > 0
+
+            # Delete the resources
+            p_client.delete(certificate_valid)
+            p_client.delete(certificate_allns_valid)
+
+    @if_test_rbac
+    @pytest.mark.parametrize("role", rbac_role_list)
+    def test_edit_certificate(self, role):
```

Comment:
Can you add a test case that the project owner/member cannot edit the namespaces certificates that are not in their project?

Reply:
Added test cases